lib/shared.js

```diff
@@ -15,192 +32,104 @@ module.exports = {
     // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-array-constructor.md
     "no-array-constructor": "off",
-    "@typescript-eslint/no-array-constructor": "error",
+    "@typescript-eslint/no-array-constructor":
+      baseStyleRules["no-array-constructor"],
 
     // Replace Airbnb 'no-empty-function' rule with '@typescript-eslint' version
     // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-empty-function.md
     "no-empty-function": "off",
-    "@typescript-eslint/no-empty-function": [
-      "error",
-      {
-        allow: ["arrowFunctions", "functions", "methods"],
-      },
-    ],
+    "@typescript-eslint/no-empty-function":
+      baseBestPracticesRules["no-empty-function"],
 
     // Replace Airbnb 'no-extra-parens' rule with '@typescript-indent' version
     // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-extra-parens.md
     "no-extra-parens": "off",
-    "@typescript-eslint/no-extra-parens": [
-      "off",
-      "all",
-      {
-        conditionalAssign: true,
-        nestedBinaryExpressions: false,
-        returnAssign: false,
-        ignoreJSX: "all", // delegate to eslint-plugin-react
-        enforceForArrowConditionals: false,
-      },
-    ],
+    "@typescript-eslint/no-extra-parens": baseErrorsRules["no-extra-parens"],
 
     // Replace Airbnb 'no-magic-numbers' rule with '@typescript-indent' version
     // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-magic-numbers.md
     "no-magic-numbers": "off",
-    "@typescript-eslint/no-magic-numbers": [
-      "off",
-      {
-        ignore: [],
-        ignoreArrayIndexes: true,
-        enforceConst: true,
-        detectObjects: false,
-      },
-    ],
+    "@typescript-eslint/no-magic-numbers":
+      baseBestPracticesRules["no-magic-numbers"],
 
     // Replace Airbnb 'no-unused-vars' rule with '@typescript-eslint' version
     // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars.md
     "no-unused-vars": "off",
-    "@typescript-eslint/no-unused-vars": [
-      "error",
-      { vars: "all", args: "after-used", ignoreRestSiblings: true },
-    ],
+    "@typescript-eslint/no-unused-vars": baseVariablesRules["no-unused-vars"],
 
     // Replace Airbnb 'no-use-before-define' rule with '@typescript-eslint' version
     // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-use-before-define.md
     "no-use-before-define": "off",
     "@typescript-eslint/no-use-before-define": [
-      "error",
-      { functions: true, classes: true, variables: true },
+      baseVariablesRules["no-use-before-define"][0],
+      {
+        ...baseVariablesRules["no-use-before-define"][1],
+        typedefs: true,
```

Comment:
I'm not sure about adding this rule. The goal of this project is to simply mirror `eslint-config-airbnb`, without colouring or adding any recommended settings. This recommended setting can be added by the user (if they want it). E.g.:

```
{
  "extends": [
    "airbnb-typescript",
    "plugin:@typescript-eslint/recommended"
  ],
}
```